src/service.rs

```diff
@@ -546,6 +576,27 @@ mod test {
             Event::LostPeer(id) => assert_eq!(id, id_0),
             e => panic!("Received unexpected event: {:?}", e),
         }
+
+        // Drop services and make sure the event channels close
+        drop(service_0);
+        let (done_tx, done_rx) = mpsc::channel();
+        let tj = thread!("Drain event channel messages", move || {
+            for _ in event_rx_0 {}
```

Comment:
Don't we know exactly at this point which remaining events to expect? (Probably none at all?)